flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/Buffer.java

```diff
@@ -87,6 +87,13 @@ public interface Buffer {
      */
     BufferRecycler getRecycler();
 
+    /**
+     * Sets the buffer's recycler.
+     *
+     * @param bufferRecycler the new buffer recycler
+     */
+    void setRecycler(BufferRecycler bufferRecycler);
```

Comment:
This is an unsafe operation, we may need add more descriptions of this method to avoid others accidentally introducing bugs. Especially for slice buffer.

Reply:
Fixed. Added docs to describe the method.